### PR TITLE
feat(protocol-designer): delete unused code

### PR DESCRIPTION
**Pull request by jerader**
closes AUTH-645

# Overview

**We are free to merge this in since the PD release will be off of the release branch!** 

This PR deletes unused code and moves a few components into the correct folder (such as `LabwareOnDeck`, `getUnusedEntities`). There are also a few redux actions/selectors/reducers that were deleted because they are no longer in use. Those include:

1. thermocycler profile step -> the redesign has it contained in a modal and keeps track of stuff via local states
2. a bunch of hints -> we deprecated most of them for the redesign. only remaining hints are for exporting a protocol with unused hardware and changing your magnetic module version
3. redux actions related to navigation -> we use react-router-dom now for navigation
4. stuff related to expanding and collapsing steps and multi-steps -> we no longer allow expanding/collapsing steps at all

Additionally, all css style sheets are officially deleted 🗑️ 

## Test Plan and Hands on Testing

Thoroughly smoke test locally. 

## Changelog

🗑️ 

## Risk assessment

medium? it deletes a bunch of code but it off of `edge`

### CODE REVIEW

**Review by ddcc4 — APPROVED** [review]
stampstampstamp

File: protocol-designer/src/ProtocolEditor.tsx
Comment (koji) [inline protocol-designer/src/ProtocolEditor.tsx:1]: question, is there any reason to do the following?

```ts
export function ProtocolEditor(): JSX.Element {
  return (
    <DndProvider backend={HTML5Backend}>
      <Box
        width="100%"
        height="100vh"
        overflow={OVERFLOW_AUTO}
        id="protocol-editor"
      >
        <PortalRoot />
        <Flex flexDirection={DIRECTION_COLUMN}>
          <HashRouter>
            <ProtocolRoutes />
          </HashRouter>
        </Flex>
      </Box>
    </DndProvider>
  )
}

```

Comment (jerader) [reply protocol-designer/src/ProtocolEditor.tsx:1]: let me see. i have a feeling it is remnants from old code where we needed to wrap the DnD provider like that. I'll check with the version you suggested.

Comment (jerader) [reply protocol-designer/src/ProtocolEditor.tsx:1]: good news, it seems to work. i'll push the change.

**Review by koji — APPROVED** [review]
thank you for doing this!

**Pull request merged**